File: csharp/raindrops/Raindrops.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Raindrops
{
    private static bool DivisibleBy(int number, int divisor)
    {
        return (number % divisor) == 0;
    }

    private static readonly Dictionary<int, string> _sounds = new Dictionary<int, string>
    {
        { 3, "Pling" },
        { 5, "Plang" },
        { 7, "Plong" }
    };

    public static string Convert(int number)
    {
        var result = string.Join("", _sounds.Keys.Select(k => DivisibleBy(number, k) ? _sounds[k] : "" ));
        return (result == "") ? number.ToString() : result;
    }
}

File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
)

type RainSound struct {
	factor int
	sound  string
}

var sounds = []RainSound{
	RainSound{3, "Pling"},
	RainSound{5, "Plang"},
	RainSound{7, "Plong"},
}

func Convert(value int) string {
	result := ""
	for _, s := range sounds {
		if value%s.factor == 0 {
			result += s.sound
		}
	}
	if result == "" {
		result = fmt.Sprintf("%d", value)
	}
	return result
}
